### PR TITLE
Updating dependencies and fixing tests

### PR DESCRIPTION
**Pull request by stulees**
- Removing specifying old protractor version, ensuring widget relies on widget-tester
- Fixing setting e2e tests due to protractor version upgrade
- Fixing other failing tests

### CODE REVIEW

**Comment by stulees** [comment]
@alex-deaconu Please review, thanks!

File: test/e2e/settings.js [hunk 1]
```diff
@@ -60,25 +60,44 @@
...
       element(by.id("save")).click();
+
+      browser.wait(function() {
+        var deferred = protractor.promise.defer();
+        element(by.id("invalid-image-error")).isDisplayed()
+          .then(function (isVisible) {
+            deferred.fulfill(isVisible);
+          });
+        return deferred.promise;
+      }, 1000);
```
Comment (alex-deaconu) [inline test/e2e/settings.js]: @stulees FYI we have some handy e2e functions similar to this in the rv-common-e2e project: https://github.com/Rise-Vision/rv-common-e2e/blob/master/lib/common/helper.js. 

Thanks.

File: package.json [hunk 2]
```diff
@@ -35,10 +35,9 @@
     "jshint-stylish": "^0.2.0",
     "mocha": "^2.1.0",
     "phantomjs": "^1.9.16",
-    "protractor": "~1.0.0",
     "run-sequence": "^0.3.2",
     "sinon": "^1.14.1",
-    "web-component-tester": "^3.3.0",
+    "web-component-tester": "*",
     "widget-tester": "git://github.com/Rise-Vision/widget-tester.git"
```
Comment (alex-deaconu) [inline package.json]: @stulees I *think* you can potentially remove all testing dependencies from this file and rely on `widget-tester` to load them:
sinon, mocha, phantomjs, chai & chai-as-promised

We've done that with Apps without any issues. Just something to keep in mind, and test out first of course. Thanks.

**Comment by alex-deaconu** [comment]
@stulees looks good :+1: . Just 2 general comments, but nothing that needs to be addressed for this PR. Thanks!

File: test/e2e/settings.js (hunk 1 above)
Comment (stulees) [inline test/e2e/settings.js]: @alex-deaconu Nice, I'll look to possibly leverage this file in a further PR for this repo, thanks.

File: package.json (hunk 2 above)
Comment (stulees) [inline package.json]: @alex-deaconu Ya good call and have previously removed a bunch already. I'll see about removing these too in the next PR, thanks.

**Pull request merged**